src/arrayMethodReduce.js

```diff
@@ -5,7 +5,15 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let result = initialValue === undefined ? this[0] : initialValue;
+    let index = initialValue === undefined ? 1 : 0;
+
+    while (index < this.length) {
+      result = callback(result, this[index], index, this);
+      index++;
+    }
```

Comment:
```suggestion
    for (let i = index; i < this.length; i++) {
      result = callback(result, this[index], index, this);
    }
```
That would be more meaningful